src/main.rs

```diff
@@ -1,49 +1,17 @@
-use std::env;
-use std::fs;
-use std::process;
-use std::error::Error;
-use std::io::{BufRead, BufReader};
-use std::path::Path;
 
+pub mod conf;
 
-struct Config{
-    path : String,
-    root : String,
-}
-
-impl Config {
-    fn new(args: &[String]) -> Result<Config, &'static str> {
-        if args.len() < 3 {
-            return Err("not enough arguments supplied to function!")
-        }
-        let path = args[1].clone();
-        let root = args[2].clone();
-        Ok(Config { path, root })
-    }
-}
-
-fn run(config : Config) -> Result<(), Box<dyn Error>>{
-    let file = fs::File::open(config.path)?;
-    let reader = BufReader::new(file);
-
-    for(_index, line) in reader.lines().enumerate(){
-        let new_path = Path::new(&config.root).join(line?);
-        println!("Created folder : {}", new_path.display());
+use std::env;
+use std::process;
 
-        fs::create_dir_all(new_path)?;
-    }
-    Ok(())
-}
+use crate::conf::Config;
 
 fn main() {
-    let args : Vec<String> = env::args().collect();
+    let args = env::args().collect::<Vec<String>>();
```

Comment:
```
let args : Vec<String> = env::args().collect()
let args = env::args().collect::<Vec<String>>();
```
What would be the difference between these two lines?

Reply:
Nothing functionally, so you might say superfluous. But I think it reads better.
Looks for _turbofish_, in Rust, it's the idea that if you have a _generic function_, like `collect` is, essentially `collect` is `collect<T>`, you can use the tubrofish `::<T>` to say what is the exact `T` you want to use.